content/chainguard/chainguard-enforce/authentication/identity-examples/enforce-aws-identity.md

```diff
@@ -0,0 +1,342 @@
+
+Now you can move on to creating the last of our Terraform configuration files, `aws.tf`.
+
+### `lambda.tf`
+
+The `lambda.tf` describes the AWS role that a Lambda function will run as.
+
+```hcl
+data "aws_iam_policy_document" "lambda" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy" "lambda" {
+  name = "AWSLambdaBasicExecutionRole"
+}
+
+resource "aws_iam_role" "lambda" {
+  name                = "aws-auth"
+  assume_role_policy  = data.aws_iam_policy_document.lambda.json
+  managed_policy_arns = [data.aws_iam_policy.lambda.arn]
+}
+```
+
+This describes an AWS role that a Lambda function will run as. The Lambda function will assume this role, and then use the Chainguard identity to interact with Chainguard resources.
+
+In `chainguard.tf`, we define a Chainguard identity that can be assumed by the AWS role:
+
+```hcl
+resource "chainguard_identity" "aws" {
+  parent_id   = chainguard_group.user-group.id
+  name        = "aws-auth-identity"
+  description = "Identity for AWS Lambda"
+
+  aws_identity {
+    aws_account         = data.aws_caller_identity.current.account_id
+    aws_user_id_pattern = "^AROA(.*):${local.lambda_name}$"
+
+    // NB: This role will be assumed so can't use the role ARN directly. We must use the ARN of the assumed role
+    aws_arn = "arn:aws:sts::${data.aws_caller_identity.current.account_id}:assumed-role/${aws_iam_role.lambda.name}/${local.lambda_name}"
+  }
+}
+```
+
+The most important part of this section is the `aws_identity` block. When the AWS role tries to assume this identity later on, it must present a token matching the `aws_account`, `aws_user_id_pattern`, and `aws_arn` specified here in order to do so.
+
+The `aws_user_id_pattern` field configures the identity to be assumable only by the AWS role with the specified name, which is itself assumed by another execution role, which we'll configure below. This role will be assumed so can't use the role ARN directly in `aws_arn`; We must used the ARN of the assumed role.
+
+The section after that looks up the `viewer` role.
+
+```
+data "chainguard_roles" "viewer" {
+  name = "viewer"
+}
+```
+
+The final section grants this role to the identity on the `example-group`.
+
+```
+resource "chainguard_rolebinding" "view-stuff" {
+  identity = chainguard_identity.aws.id
+  group	= chainguard_group.example-group.id
+  role 	= data.chainguard_roles.viewer.items[0].id
+}
+```
+
+After defining these resources, there are some other resources in the example directory that build and deploy a Lambda function that assumes the identity. We'll describe that code in the next section.
+
+After defining these resources, your Terraform configuration will be ready. Now you can run a few `terraform` commands to create the resources defined in your `.tf` files.
+
+### `lambda.tf`
```

Comment:
I think things got a little twisted here - we can delete this whole last section, right?